tests/roles/ceph_migrate/README.md

```diff
@@ -0,0 +1,143 @@
+Role Name
+=========
+
+The goal of this role is to `externalize` a TripleO deployed Ceph cluster. It is
+intended to be run in CI, where a `multinode` TripleO environment with deployed
+Ceph exists.
+It **does not** migrate OSDs, hence no data movement is involved.
+It assumes that both `source` and `target` nodes are part of the Ceph cluster,
+and the high level procedure does the following:
+
+1. It Loads the information from the existing Ceph cluster (deployed daemons,
+   status of the cluster, hostmap, services, monmap, osdmap)
+2. It Validates the retrieved information (the cluster is healthy and it's in a
+   valid state), and fail is one of the defined conditions is not met
+3. It starts performing the migration with the following order:
+   - Ceph Monitoring Stack
+   - Ceph MDS
+   - Ceph RGW
+   - Ceph MGR
+   - Ceph MON
+4. It dumps the result of the performed migration
+
+Requirements
+------------
+
+In order to be able to run the playbook associated with this role, a `3+3`
+topology is at least required (3 Controller, 3 HCI/CephStorage nodes).
+If such topology is not available, it's still possible to run the playbook and
+perform (as an example) the migration of a single node using tags and passing
+overrides.
+To migrate only a subset of Mon/Mgr, it is possible to customize the list of
+`decomm_nodes` and `target_nodes`:
+
+```
+decomm_nodes:
+  - controller-0
+target_nodes:
+  - cephstorage-0
```

Comment:
who/what provides this node? I mean, the tripleo/source deployment will have 3 control 3 compute.

Reply:
I added an additional sentence to clarify how, in the example, the `hostname(s)` of the nodes are gathered. In particular, the example shows how to pass overrides to decommission a single node, moving daemons (mons in particular) from src to the dest. This might be useful considering that we do not have a `3+3` right now, but we might want to try a single `mon` migration to prove that the flow is safe.

Reply:
@marios as a follow up of this work we need to plug this playbook to our CI, which is where we might need a "glue" code that generates the overrides and runs the playbook. Given that we know in advance the topology, and the resulting hosts data is predictable in CI, this might just be something we pass from a config job.

Reply:
ack lets talk revisit this when we are ready. Joan is working through the ceph wire up (for 3ctrl/1compute) currently so that is a prerequisite